File: data/src/main/java/com/github/chojmi/inspirations/data/entity/photos/PhotoCommentsEntityImpl.java
package com.github.chojmi.inspirations.data.entity.photos;

import android.os.Parcelable;

import com.github.chojmi.inspirations.domain.entity.photos.PhotoCommentsEntity;
import com.google.auto.value.AutoValue;
import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.SerializedName;

import java.util.List;

@AutoValue
public abstract class PhotoCommentsEntityImpl implements Parcelable, PhotoCommentsEntity {
    public static TypeAdapter<PhotoCommentsEntityImpl> typeAdapter(Gson gson) {
        return new AutoValue_PhotoCommentsEntityImpl.GsonTypeAdapter(gson);
    }

    @SerializedName("comment")
    public abstract List<CommentEntityImpl> getComments();

    @SerializedName("photo_id")
    public abstract String getPhotoId();
}

File: data/src/main/java/com/github/chojmi/inspirations/data/entity/photos/PhotoCommentsEntityImpl.kt
package com.github.chojmi.inspirations.data.entity.photos

import com.github.chojmi.inspirations.domain.entity.photos.PhotoCommentsEntity
import com.google.gson.annotations.SerializedName

data class PhotoCommentsEntityImpl(private @SerializedName("comment") val comments: List<CommentEntityImpl>,
                                   private @SerializedName("photo_id") val photoId: String) : PhotoCommentsEntity<CommentEntityImpl> {
    override fun getComments(): List<CommentEntityImpl> {
        return comments
    }

    override fun getPhotoId(): String {
        return photoId
    }
}